include/envoy/upstream/load_balancer.h

```diff
@@ -43,6 +43,27 @@ class LoadBalancerContext {
    * balancing.
    */
   virtual const Http::HeaderMap* downstreamHeaders() const PURE;
+
+  /**
+   * Called to retrieve an optional pre priority selection host predicate. If present, this will be
+   * used to rebuild the per priority load data which only includes hosts that match the predicate.
+   *
```

Comment:
Also, please document what the parameters to the returned function are, e.g. is the first parameter priority level?